src/v/wasm/transform_module.cc

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "transform_module.h"
+
+#include "bytes/iobuf_parser.h"
+#include "model/compression.h"
+#include "model/record.h"
+#include "model/record_batch_types.h"
+#include "model/timestamp.h"
+#include "utils/named_type.h"
+#include "utils/vint.h"
+#include "vassert.h"
+#include "wasm/ffi.h"
+#include "wasm/logger.h"
+
+#include <algorithm>
+#include <exception>
+#include <optional>
+#include <stdexcept>
+#include <vector>
+namespace wasm {
+
+model::record_batch transform_module::for_each_record(
```

Comment:
i'm starting to get the impression that these `_module` files are like an interface of some sort to wasm. here for iterating over records in a batch and earlier for the schema registry?

without really knowing what all the words in the commit message mean i was confused. maybe a short sentence linking things together might be helpful for other reviewer.

Reply:
>Did the comment on the class in the header help? 

yeh thanks